karapace/backup/api.py

```diff
@@ -449,7 +468,7 @@ def restore_backup(
     # We set up an ExitStack context, so that we can enter the producer context only
     # after processing a RestoreTopic instruction.
     with contextlib.ExitStack() as stack:
-        producer = None
+        producer: BackupRestorationProducer | None = None
```

Comment:
thought (non-blocking): It's surprising that this needs an annotation now, but didn't before.

Reply:
it's not necessary, i just sometimes use type annotations to be more explicit about what it's being created, in this case i wanted to be explicit that here we're using a `BackupRestorationProducer` and not a normal producer

Reply:
If it's not necessary, I'd skip it and just rely on inference. (opinionated, so feel free to ignore! 🙂)